core/src/main/java/hudson/lifecycle/SolarisSMFLifecycle.java

```diff
@@ -38,7 +38,7 @@
      */
     @Override
     public void restart() throws IOException, InterruptedException {
-        Jenkins h = Jenkins.getInstanceOrNull();
+        Jenkins h = Jenkins.getInstanceOrNull(); // TODO confirm safe to assume non-null and use getInstance()
```

Comment:
Probably here for idempotency: two concurrent calls to `restart` separated by a slight delay could result in the second seeing `Jenkins.instance == null`.